invenio_mail/ext.py

```diff
@@ -67,5 +67,11 @@ def init_config(self, app):
     def print_email(self, message, app):
         """Print mail to stream."""
         with self._lock:
-            self.stream.write("{0}\n{1}\n".format(message.as_string(), '-'*79))
-            self.stream.flush()
+            invenio_mail = app.extensions['invenio-mail']
```

Comment:
It is true that there are several instances of ```invenio-mail```. I think that the easiest change is something like:
```python
invenio_mail = app.extensions['invenio-mail']
with invenio_mail._lock:

Because the instance in app.extensions is always the same.